src/main/java/slimeknights/tconstruct/smeltery/recipe/ICastingInventory.java

```diff
@@ -0,0 +1,108 @@
+package slimeknights.tconstruct.smeltery.recipe;
+
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.fluid.Fluid;
+import net.minecraft.inventory.IInventory;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.fluids.FluidStack;
+
+import java.util.Set;
+
+/* TODO: move the required methods to base interface */
+public interface ICastingInventory extends IInventory {
```

Comment:
Should be able to extend the new `ISingleItemInventory`, which will do all the base methods for you (solving my TODO when I sent you this interface). You will only need to add the `getFluid()` method then, as `ISingleItemInventory` gives you `getStack()`